geowombat/ml/classifiers.py

```diff
@@ -234,30 +225,50 @@ def fit(
             >>> labels = gpd.read_file(l8_224078_20200518_polygons)
             >>> labels['lc'] = le.fit(labels.name).transform(labels.name)
             >>>
-            >>> # Use a data pipeline
+            >>> # Use supervised classification pipeline
             >>> pl = Pipeline([('scaler', StandardScaler()),
             >>>                ('pca', PCA()),
             >>>                ('clf', GaussianNB())])
             >>>
             >>> with gw.open(l8_224078_20200518) as src:
-            >>>   X, clf = fit(src, labels, pl, grid_search=True, col='lc')
+            >>>   X, clf = fit(src, pl, labels, col='lc')
+
+            >>> # Fit an unsupervised classifier
+            >>> cl = Pipeline([('pca', PCA()),
+            >>>                ('cst', KMeans()))])
+            >>> with gw.open(l8_224078_20200518) as src:
+            >>>    X, clf = fit(src, cl_w_feat)
         """
 
-        data = self._prepare_labels(data, labels, col, targ_name)
-        X, Xna = self._prepare_predictors(data, targ_name)
-        clf = self._prepare_classifiers(clf)
+        if clf._estimator_type == "clusterer":
 
-        if grid_search:
-            clf = self.grid_search_cv(clf)
+            data = self._add_time_dim(data)
+            X = self._stack_it(data)
+            clf = self._prepare_classifiers(clf)
 
-        # TODO: should we be using lazy=True?
-        y = Target(
-            coord=targ_name,
-            transform_func=LabelEncoder().fit_transform,
-            dim=targ_dim_name,
-        )(Xna)
+            # TO DO: Validation checks
+            # check_array(X)
+
+            clf.fit(X)
+
+        else:
 
-        clf.fit(Xna, y)
+            data = self._prepare_labels(data, labels, col, targ_name)
+            X, Xna = self._prepare_predictors(data, targ_name)
+            clf = self._prepare_classifiers(clf)
+
+            # TODO: should we be using lazy=True?
+            y = Target(
+                coord=targ_name,
+                transform_func=LabelEncoder().fit_transform,
+                dim=targ_dim_name,
+            )(Xna)
+
+            # TO DO: Validation checks
+            # Xna, y = check_X_y(Xna, y)
+
+            clf.fit(Xna, y)
+            self.classes_ = np.unique(y)
```

Comment:
Do we have a default for `classes_`? Better yet, make it a `property` with a `setter`?